src/inner_product_round.rs

```diff
@@ -117,9 +115,19 @@ where
     }
 
     /// Calculate the inner product, updating 'self' for each round
-    pub fn inner_product<T: RngCore + CryptoRng>(&mut self, rng: &mut T) -> Result<(), ProofError> {
+    pub(crate) fn inner_product<T: RngCore + CryptoRng>(&mut self, rng: &mut T) -> Result<(), ProofError> {
+        // Ensure that vector lengths are still consistent, just in case
         let mut n = self.gi_base.len();
-        let extension_degree = self.extension_degree as usize;
+        if !(self.hi_base.len() == n && self.ai.len() == n && self.bi.len() == n && self.y_powers.len() >= n + 2) {
+            return Err(ProofError::InvalidArgument(
+                "Vector length for inner product round".to_string(),
+            ));
+        }
+        let extension_degree = ExtensionDegree::try_from_size(self.g_base.len())? as usize;
+        if extension_degree != self.alpha.len() {
+            return Err(ProofError::InvalidLength("Inconsistent extension degree".to_string()));
+        }
```

Comment:
I would keep the assignment of `extension_degree` in the constructor as it should never change
```suggestion
        let extension_degree = self.extension_degree as usize;
        if !(extension_degree == self.alpha.len() && extension_degree == self.g_base.len()) {
            return Err(ProofError::InvalidLength("Inconsistent extension degree".to_string()));
        }
```